docs/source/preview_custom_datasets.md

```diff
@@ -0,0 +1,52 @@
+# Extend Preview to Custom Datasets
+
+When creating a custom dataset, if you wish to enable data preview for that dataset, you must implement a `preview()` function within the custom dataset class. Kedro-Viz currently supports previewing tables, Plotly charts, images, and JSON objects.
+
+The return type of the `preview()` function should match one of the following types, as defined in the `kedro-datasets` source code (_typing.py file):
```

Comment:
Any chance to add a link to such file?

Reply:
is the github link fine ? - https://github.com/kedro-org/kedro-plugins/blob/main/kedro-datasets/kedro_datasets/_typing.py 

I can't seem to find docs source code link for _typing.py

Reply:
Yeah this is not documented, so a link to the source code is fine for now.